app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -176,3 +184,34 @@ class CiYaml {
     }
   }
 }
+
+/// Class to expose the PinnedVersion method for testing.
+class DependencyValidator {
+  /// Dependency is guaranteed to be non empty as it must be found before this
```

Comment:
Nit: Let's update the doc a bit to follow the style: https://dart.dev/guides/language/effective-dart/documentation. Like, use sentence, capital for the first word, etc.